util/fileinfo.go

```diff
@@ -0,0 +1,42 @@
+package util
+
+import (
+	"fmt"
+)
+
+// FileInfo represents a configuration source.  It is intended to help the user
+// understand how the system acquired its configuration, and to aid in tracking
+// down errors in configuration files.
+type FileInfo struct {
+	Path   string    // Path of configuration file.
+	Parent *FileInfo // File that imported this configuration file, if any.
+}
+
+// ImportString creates a string describing the import hierarchy of the given
+// FileInfo.  It should be called on the *parent* of the file of interest.
+func (fi *FileInfo) ImportString() string {
+	s := ""
+
+	first := true
+	for fi != nil {
+		if !first {
```

Comment:
if `first` starts with `true` will the contents of this `if` ever execute?

Reply:
Ah, you are right - the `first = false` statement should be outside the `if`.